examples/contracts/examples/deploy_from_contract.rs

```diff
@@ -36,37 +36,46 @@ async fn main() -> Result<()> {
 
     // Set up wallet
     let wallet: LocalWallet = anvil.keys()[0].clone().into();
+    let alice = anvil.addresses()[0];
 
     // Create a provider with a signer.
     let http = anvil.endpoint().parse()?;
     let provider = ProviderBuilder::new()
+        .with_recommended_layers()
         .signer(EthereumSigner::from(wallet))
         .on_client(RpcClient::new_http(http));
 
     println!("Anvil running at `{}`", anvil.endpoint());
 
-    // Get the base fee for the block.
-    let base_fee = provider.get_gas_price().await?;
-
     // Deploy the contract.
     let contract_builder = Counter::deploy_builder(&provider);
-    let estimate = contract_builder.estimate_gas().await?;
-    let contract_address =
-        contract_builder.gas(estimate).gas_price(base_fee).nonce(0).deploy().await?;
+    let contract_address = contract_builder
+        .from(alice)
+        .map(|mut tx| {
+            tx.set_chain_id(anvil.chain_id());
+            tx
+        })
```

Comment:
Expected as a method directly accessible on `contract_builder`

Reply:
The default case will likely be resolved by the new `FillTxLayer` proposal: https://github.com/alloy-rs/alloy/pull/374 but generally still a good method to have direct access to I think.